server/src/db/replay.rs

```diff
@@ -2568,6 +2598,72 @@ mod tests {
         .await;
     }
 
+    #[tokio::test]
+    async fn replay_delete() {
+        ReplayTest {
+            steps: vec![
+                Step::Ingest(vec![TestSequencedEntry {
+                    sequencer_id: 0,
+                    sequence_number: 0,
+                    lp: "table_1,tag_partition_by=a bar=10 10",
+                }]),
+                Step::Await(vec![Check::Query(
+                    "select * from table_1 order by bar",
+                    vec![
+                        "+-----+------------------+--------------------------------+",
+                        "| bar | tag_partition_by | time                           |",
+                        "+-----+------------------+--------------------------------+",
+                        "| 10  | a                | 1970-01-01T00:00:00.000000010Z |",
+                        "+-----+------------------+--------------------------------+",
+                    ],
+                )]),
+                Step::MakeWritesPersistable,
+                Step::Persist(vec![("table_1", "tag_partition_by_a")]),
+                Step::Delete(vec![TestDelete {
+                    sequencer_id: 0,
+                    sequence_number: 1,
+                    table_name: None,
+                    predicate: DeletePredicate {
+                        range: TimestampRange { start: 0, end: 11 },
```

Comment:
What do you think about increasing this range to be to 20 (so it also would apply to the data that was ingested *after* it) to demonstrate the delete is applied correctly to only the data that was ingested prior ot it?

```suggestion
                        range: TimestampRange { start: 0, end: 20 },
```